src/records/RecYAMLDecoder.cc

```diff
@@ -146,15 +146,16 @@ ParseRecordsFromYAML(YAML::Node root, RecYAMLNodeHandler handler, bool lock /*fa
 
   swoc::Errata errata;
   if (YAML::NodeType::Map != root.Type()) {
-    return swoc::Errata(ERRATA_NOTE, "Node is expected to be a map, got '{}' instead.", root.Type());
+    return swoc::Errata(ERRATA_ERROR, "Node is expected to be a map, got '{}' instead.", root.Type());
   }
 
   if (auto ts = root[RECORD_YAML_ROOT_STR]; ts.size()) {
     for (auto &&n : ts) {
       detail::flatten_node({n.first, n.second, CONFIG_RECORD_PREFIX}, handler, errata);
     }
   } else {
-    return swoc::Errata(ERRATA_NOTE, "'records' root key not present or no fields to read.");
+    return swoc::Errata(ERRATA_ERROR, "'{}' root key not present or no fields to read. Default values will be used",
```

Comment:
how do we differentiate warnings and errors? if TS still runs but without the expected configurations, is that just a warning or error? it was a note before so I'm just curious the jump to error

Reply:
The only change will be the text from `Note` to `Error`, the idea is to make it more relevant in the log as not having the `records` node in the config should call someones attention.

Reply:
Right, I agree Note is not sufficient. Even though Error should indicate that something won't work or is broken, not using the configured records.yaml is bad

Reply:
I was thinking it should be somewhat consistent with your above comment. In that comment, when the file doesn't exist, it uses the default values and uses a warning comment. Here, the records root key doesn't exist so it uses the default values too but it's an error comment.

Reply:
so, to differentiate, which I should have done in the first comment, sorry my bad. This will show something like this:


```
[Jul 16 16:06:30.658] traffic_server WARNING: We have found the following issues when reading the records node:
 Error:   'records' root key not present or no fields to read.
  Warn: Ignoring field 'invalid_record' at Line 114. Not registered and Unknown tag type '?'
 ```
So, note the main message:
```
WARNING: We have found the following issues when reading the records node:
```
and then, the subsequently ones which are part of the first warning, but they are error,warning and note in the context of the first warning, and should be read as messages with different severities(Err,WArn,Note, etc)

Reply:
Ah I see, thank you for explaining!

Reply:
I think we need to find a way to make the `Errata` and the regular log levels to work together, to avoid somehow misleading level of messages like this one can be, this was Alan's idea I guess somehow. I'll try to spend time in the future to work around this.

Thanks for reviewing this @serrislew